File: getting staed/power.c
#include <stdio.h>
#include <math.h>

double power(double base, double exp)
{
    double result = 1.0;
    if (exp > 0.0)
    {
        while (exp > 0.0)
        {
            result = result * base;
            exp--;
        }
        return result;
    }
    if (exp < 0.0)
    {
        while (exp < 0.0)
        {
            result = result / base;
            exp++;
        }
        return result;
    }
    return 1;
}

int main()
{
    int base = 2.0, exp = -8.0;
    printf("The power claculated as %lf", power(base, exp));
    return 0;
}

File: getting staed/power.java
public class power{
    public static void main(String[] args) {
        int base=10;
        int pow=3;
        int ans=1;
        while(pow>0){
            if((pow&1)==1)
            ans*=base;
            base*=base;
           pow= pow>>1;
        }
        System.out.println(ans);
    }
}